liquidctl/config/load.py

```diff
@@ -0,0 +1,53 @@
+import logging
+import os
+import sys
+
+def get_config_files(file=None, appname='liquidctl', **kwargs):
+    """
+    This will get all the potential file paths on the system for where the config file
+    can be
+
+    This will also handle the -f overrides
+    """
+    files = []
+    if file:
+        files.append(file)
+
+    if sys.platform == 'win32':
+        files.append(os.path.join(os.getenv('APPDATA'), appname, 'config.toml'))
+    elif sys.platform == 'darwin':
+        files.append(os.path.expanduser(os.path.join('~/Library/Application Support', appname, 'config.toml')))
+        files.append(os.path.expanduser(os.path.join('~', f'.{appname}.toml')))
+    elif sys.platform == 'linux':
+        XDG_CONFIG_HOME = os.getenv('XDG_CONFIG_HOME')
+        XDG_CONFIG_DIRS = os.getenv('XDG_CONFIG_DIRS')
+
+        # treat all other platforms as *nix and conform to XDG basedir spec
+        if XDG_CONFIG_HOME:
+            files.append(os.path.join(XDG_CONFIG_HOME, appname, 'config.toml'))
+
+        files.append(os.path.expanduser(os.path.join('~', '.config', appname, 'config.toml')))
+        files.append(os.path.expanduser(os.path.join('~', f'.{appname}.toml')))
+
+        if XDG_CONFIG_DIRS:
+            files.append(os.path.join(XDG_CONFIG_DIRS, appname, 'config.toml'))
+    else:
+        files.append(os.path.expanduser(os.path.join('~', f'.{appname}.toml')))
+
+    return files
+
+def filter_config_files(files=[], ignore_config=False, **kwargs):
+    """
+    This will take a list of file paths and return the first one that exists from the list.
+    If the ignore config option is set then it will return None even if files do exist.
+    """
```

Comment:
Alternatively, we might want to merge the sections from the various files that we found, allowing the user config file to override _sections_ of the global/system config.

_(I think this is what I was thinking of when I messed up above)._

Reply:
> @CaseySJ, do you have any idea on what we should do here?

You're right. On macOS there are two places where applications can store any app-specific files:

- For a particular user: `~/Library/Application Support/<app name>/file(s)`
- For all users: `/Library/Application Support/<app name>/file(s)`

Reply:
Thanks @CaseySJ !

Reply:
> or all users:.... 

I will add that to the mac location checking then.

Reply:
> Alternatively, we might want to merge the sections from the various files that we found....

Im not sure how I feel about that, it makes sense but at the same time I have recently been working with spring boot and their config files work like that where they can override specific sections. Thats has been suuuuuuper confusing to work with to figure out what config file is causing things to go wrong because you might forget about what is in which file or where all the different locations are.

Reply:
I agree that it can be confusing.  But it's fairly common on Unix programs, so it may be expected.

Also, I think it's the only sensible way to have both system wide and user specific settings.  If the presence of a user specific config file makes the system wide one be ignored, this may also cause some confusion.

One thing that may help here (regardless of which approach is taken) is that we provide some information on `--debug`:

```
INFO: ~/.config/liquidctl/config.toml overrides /etc/liquidctl/config.toml
INFO: section dev1.<foo>.<bar> in ~/.config/liquidctl/config.toml overrides previous values
```